### PR TITLE
Bug 1181687 - Use real Promises in MockLazyLoader, r=ferjm

### PR DESCRIPTION
**Pull request by michielbdejong**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b49ec6d948289ebfc76400937e96c90835c69791

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bca00c1a3eb81714d97047369f3ced4af2e68bc0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c4ebfa23ff9abb78852c9c81099ee151922208a5

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=245cc974f6ae0ba5d73fc084895e7cf607324fbf

File: shared/test/unit/mocks/mock_lazy_loader.js [hunk 1]
```diff
@@ -1,19 +1,15 @@
 'use strict';
 /* exported MockLazyLoader */
 
+// Our MockLazyLoader is better than the shared one because we actually use a
```
Comment (sfoster) [inline shared/test/unit/mocks/mock_lazy_loader.js]: Nonsensical comment? This *is* the shared MockLazyLoader

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=08bc2a62efaee7736ec28d6b8fed66d00dc38d20

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3fb5e62a81e3e13da46e51f45f3f0313bf1423b6

**Comment by michielbdejong** [comment]
As per https://bugzilla.mozilla.org/show_bug.cgi?id=1181687#c7, set reviewer to @ferjm who is a system app peer.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7d2ddaa951deb264c43461afdb5d2e8f1cbbdedc

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1390d10cc62ef87ec45ed926ae504abf8ce39819

**Comment by michielbdejong** [comment]
Weird, that test is passing when I run it locally.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1fe173ed6c84c5698551b650d284d4ff67cbce54

**Pull request merged**